src/functions-and-arrays.js

```diff
@@ -1,73 +1,431 @@
 // Iteration #1: Find the maximum
 
+var maxOfTwoNumbers = [5, 7];
+
+function findMax(array) {
+  var max = -Infinity;
+
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] > max) {
+      max = array[i];
+    }
+  }
+  return max;
+}
+
+var result = findMax([5, 7]);
+
+console.log("result :", result);
+
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+/* 
+I don't fully understand:
+You can use rest parameter (...) syntax 
+allows us to represent an indefinite number of arguments as an array.*/
+function findLongestWord(...strs) {
+  return strs.sort(function(a, b) {
+    return b.length - a.length;
+  })[0];
+}
+console.log(
+  findLongestWord(
+    "mystery",
+    "brother",
+    "aviator",
+    "crocodile",
+    "pearl",
+    "orchard",
+    "crackpot"
+  )
+);
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+var sumNumbers = numbers.reduce(function(a, b) {
+  return a + b;
+}, 0);
+console.log(sumNumbers);
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+var sumNumbers = numbersAvg.reduce(function(a, b) {
+  return a + b;
+}, 0);
+console.log(sumNumbers / 8);
 
 // Level 2: Array of strings
-const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+const wordsArr = [
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
+];
+// I'm not sure about this
+averageWordLength = wordsArr.join("").length / wordsArr.length;
```

Comment:
variable is not declared. Also, this should have been a function as instructed in the repo.